src/PythonExtensionGen.cpp

```diff
@@ -295,103 +288,137 @@ HALIDE_PYTHON_EXPORT PyObject* PyInit_)INLINE_CODE";
+            indent.indent -= 2;
+            dest << indent << "}\n";
+        }  // else Python already converted this.
     }
-    dest << "    int result;\n";
-    dest << "    Py_BEGIN_ALLOW_THREADS\n";
+    dest << "\n";
     // Mark all input buffers as having a dirty host, so that the Halide call will
     // do a lazy-copy-to-GPU if needed.
     for (size_t i = 0; i < args.size(); i++) {
         if (args[i].is_buffer() && args[i].is_input()) {
-            dest << "    buffer_" << arg_names[i] << ".set_host_dirty();\n";
+            dest << indent << "b_" << arg_names[i] << ".halide_buf.set_host_dirty();\n";
         }
     }
-    dest << "    result = " << f.name << "(";
+    dest << indent << "int result;\n";
+    dest << indent << "Py_BEGIN_ALLOW_THREADS\n";
+    dest << indent << "result = " << f.name << "(\n";
+    indent.indent += 2;
     for (size_t i = 0; i < args.size(); i++) {
-        if (i > 0) {
-            dest << ", ";
-        }
         if (args[i].is_buffer()) {
-            dest << "&buffer_" << arg_names[i];
+            dest << indent << "&b_" << arg_names[i] << ".halide_buf";
         } else {
-            dest << "py_" << arg_names[i];
+            dest << indent << "py_" << arg_names[i] << "";
+        }
+        if (i < args.size() - 1) {
+            dest << ",";
         }
+        dest << "\n";
     }
-    dest << ");\n";
-    dest << "    Py_END_ALLOW_THREADS\n";
+    indent.indent -= 2;
+    dest << indent << ");\n";
+    dest << indent << "Py_END_ALLOW_THREADS\n";
     // Since the Python Buffer protocol is host-memory-only, we *must*
     // flush results back to host, otherwise the output buffer will contain
     // random garbage. (We need a better solution for this, see https://github.com/halide/Halide/issues/6868)
     for (size_t i = 0; i < args.size(); i++) {
         if (args[i].is_buffer() && args[i].is_output()) {
-            dest << "    if (result == 0) result = halide_copy_to_host(nullptr, &buffer_" << arg_names[i] << ");\n";
+            dest << indent << "if (result == 0) result = halide_copy_to_host(nullptr, &b_" << arg_names[i] << ".halide_buf);\n";
         }
     }
-    release_buffers();
-    dest << R"INLINE_CODE(
-    if (result != 0) {
-        /* In the optimal case, we'd be generating an exception declared
-         * in python_bindings/src, but since we're self-contained,
-         * we don't have access to that API. */
-        PyErr_Format(PyExc_ValueError, "Halide error %d", result);
-        return nullptr;
-    }
-    Py_INCREF(Py_True);
-    return Py_True;
-)INLINE_CODE";
+    dest << indent << "if (result != 0) {\n";
+    indent.indent += 2;
+    dest << indent << "PyErr_Format(PyExc_ValueError, \"Halide error %d\", result);\n";
+    dest << indent << "return nullptr;\n";
+    indent.indent -= 2;
+    dest << indent << "}\n";
+    dest << "\n";
+
+    dest << indent << "Py_INCREF(Py_None);\n";
+    dest << indent << "return Py_None;\n";
```

Comment:
Is there a reason we return `None` now, as opposed to `True`?

Reply:
Eh, maybe overthinking it? Conceptually, if it were written in Python, `None` would be the 'right' return value (since in C++ the return value is simply an error code, which is handled by exceptions here). Returning `True` isn't wrong per se -- it will likely be ignored by the caller -- but `None` seems right-er.